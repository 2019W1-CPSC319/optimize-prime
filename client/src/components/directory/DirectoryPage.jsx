import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
import {
  Button,
  Tab,
  Tabs,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper
} from '@material-ui/core';

import { connect } from 'react-redux';
import * as candidateSelectors from '../../selectors/CandidateSelectors';
import * as interviewerSelectors from '../../selectors/InterviewerSelectors';

const styles = theme => ({
  title: {
    fontWeight: 'normal',
    marginLeft: '30px',
  },
  flex: {
    display: 'flex',
    [theme.breakpoints.down('sm')]: {
      display: 'block',
    },
  },
  header: {
    justifyContent: 'space-between',
  },
  directory: {
    width: '100%',
    overflowX: 'auto',
  },
  table: {
    minWidth: 650,
  },
  delete: {
    color: '#ff454e',
    borderColor: '#ff454e',
    '&:hover': {
      backgroundColor:
        'rgba(255, 69, 78, 0.2)'
    }
  }
});

class DirectoryPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: 0,
      candidates: [],
      interviewers: []
    };
  }

  // get all candidates and interviewers on page load
  componentDidMount() {
    const { fetchInterviewers, fetchCandidates } = this.props.actions;
    fetchInterviewers()
    fetchCandidates()
  }

  render() {
    const { classes } = this.props;
    const { candidates, interviewers } = this.props;


    function handleChange(e, value) {
      this.setState({ value: value });
    }
    
    function Directory(props) {
      const { label, rows } = props
      return <Paper className={classes.directory}>
        <Table className={classes.table} aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>{label} Name</TableCell>
              <TableCell>Email</TableCell>
              <TableCell>Phone</TableCell>
              <TableCell></TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map(row => (
              <TableRow key={row.id}>
                <TableCell component="th" scope="row">
                  {row.firstName + " " + row.lastName}
                </TableCell>
                <TableCell>{row.email}</TableCell>
                <TableCell>{row.phone}</TableCell>
                <TableCell><Button variant="outlined" className={classes.delete}>Delete</Button></TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </Paper>;
    }

    return <div>
      <div className={classes.header}>
        <h1 className={classes.title}>Directory</h1>
      </div>
      <Paper style={{ margin: '40px 10px auto' }} square>
        <Tabs
          value={this.state.value}
          indicatorColor="primary"
          textColor="primary"
          onChange={handleChange.bind(this)}
          aria-label="disabled tabs example"
        >
          <Tab label="Candidate" />
          <Tab label="Employee" />
        </Tabs>
        <div hidden={this.state.value !== 0}><Directory label="Candidate" rows={candidates} /></div>
        <div hidden={this.state.value !== 1}><Directory label="Employee" rows={interviewers} /></div>
      </Paper>
    </div>;
  }
}

const mapStateToProps = state => ({
  candidates: candidateSelectors.getCandidates(state),
  interviewers: interviewerSelectors.getInterviewers(state)
})

export default withStyles(styles)(connect(mapStateToProps)(DirectoryPage));
