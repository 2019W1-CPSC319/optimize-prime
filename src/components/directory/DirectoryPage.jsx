import React, { Component } from 'react';
import { withStyles } from '@material-ui/core/styles';
// import VisibilityOffIcon from '@material-ui/icons/VisibilityOff';
import {
  Button,
  Tab,
  Tabs,
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Paper
} from '@material-ui/core';

const styles = theme => ({
  title: {
    fontWeight: 'normal',
    marginLeft: '30px',
  },
  flex: {
    display: 'flex',
    [theme.breakpoints.down('sm')]: {
      display: 'block',
    },
  },
  header: {
    justifyContent: 'space-between',
  },
  directory: {
    width: '100%',
    overflowX: 'auto',
  },
  table: {
    minWidth: 650,
  },
  delete: {
    color: '#ff454e',
    borderColor: '#ff454e',
    '&:hover': {
      backgroundColor:
        'rgba(255, 69, 78, 0.2)'
    }
  }
});

class DirectoryPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      value: 0
    };
  }

  render() {
    const { classes } = this.props;

    function handleChange(e, value) {
      this.setState({ value: value });
    }

    function createData(name, job, username, password) {
      return { name, job, username, password };
    }

    function Directory(props) {
      const { label, rows } = props
      return <Paper className={classes.directory}>
        <Table className={classes.table} aria-label="simple table">
          <TableHead>
            <TableRow>
              <TableCell>{label} Name</TableCell>
              <TableCell>{label === 'Candidate' ? 'University Name' : 'Job Title'}</TableCell>
              <TableCell>User Name</TableCell>
              <TableCell></TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {rows.map(row => (
              <TableRow key={row.name}>
                <TableCell component="th" scope="row">
                  {row.name}
                </TableCell>
                <TableCell>{row.job}</TableCell>
                <TableCell>{row.username}</TableCell>
                <TableCell><Button variant="outlined" className={classes.delete}>Delete</Button></TableCell>
              </TableRow>
            ))}
          </TableBody>
        </Table>
      </Paper>;
    }

    const candidates = [
      createData('Chris Evans', 'University of British Columbia', 'c.evans@gmail.com', '********'),
      createData('Robert Downey Jr.', 'Simon Fraser University', 'r.downey@gmail.com', '********'),
      createData('Scarlet Johansen', 'British Columbia Institue of Technology', 's.johansen@gmail.com', '********'),
      createData('Tom Holland', 'University of Waterloo', 't.holland@gmail.com', '********'),
      createData('Benedict Cumberbatch', 'University of Victoria', 'b.cumberbatch@gmail.com', '********'),
    ];

    const employees = [
      createData('Captain America', 'Senior Project Manager', 'c.america@galvanize.com', '********'),
      createData('Iron man', 'UX Designer', 'i.man@galvanize.com', '********'),
      createData('Black Widow', 'Junior Software Developer', 'b.widow@galvanize.com', '********'),
      createData('Spider man', 'Business Analyst', 's.man@galvanize.com', '********'),
      createData('Dr. Strange', 'Senior Project Owner', 'd.strange@galvanize.com', '********'),
    ];

    return <div>
      <div className={classes.header}>
        <h1 className={classes.title}>Directory</h1>
      </div>
      <Paper style={{ margin: '40px 10px auto' }} square>
        <Tabs
          value={this.state.value}
          indicatorColor="primary"
          textColor="primary"
          onChange={handleChange.bind(this)}
          aria-label="disabled tabs example"
        >
          <Tab label="Candidate" />
          <Tab label="Employee" />
        </Tabs>
        <div hidden={this.state.value !== 0}><Directory label="Candidate" rows={candidates} /></div>
        <div hidden={this.state.value !== 1}><Directory label="Employee" rows={employees} /></div>
      </Paper>
    </div>;
  }
}

export default withStyles(styles)(DirectoryPage);
