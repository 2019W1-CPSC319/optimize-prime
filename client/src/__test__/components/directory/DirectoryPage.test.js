import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import DirectoryPage from '../../../components/directory/DirectoryPage';

describe('Directory Page - Render', () => {
  let wrapper;
  const classes = {};
  const actions = {};

  beforeEach(() => {
    wrapper = shallow(<DirectoryPage
      classes={classes}
      actions={actions}
    />);
  });

  it('render the component', () => {
    expect(wrapper.length).toEqual(1);
  });

  it('match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
