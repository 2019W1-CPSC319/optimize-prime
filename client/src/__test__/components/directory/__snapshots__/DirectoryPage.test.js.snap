// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Page - Render match snapshot 1`] = `
<DirectoryPage
  actions={Object {}}
  classes={
    Object {
      "addButton": "DirectoryPage-addButton-4",
      "header": "DirectoryPage-header-2",
      "tableContainer": "DirectoryPage-tableContainer-3",
      "title": "DirectoryPage-title-1",
    }
  }
/>
`;
